### PR TITLE
Update commands.json file

### PR DESCRIPTION
**Pull request by ashtul**

### CODE REVIEW

File: commands.json [hunk 1]
```diff
@@ -30,5 +282,878 @@
...
+      {
+        "name": "withscores",
+        "type": "enum",
+        "optional": true,
+        "enum": [
+          "WITHSCORES"
+        ]
+      },
+      {
+        "name": "withpatloads",
```
Comment (oshadmi) [inline commands.json:679]:
```suggestion
        "name": "withpayloads",
```

File: commands.json [hunk 2]
```diff
@@ -30,5 +282,878 @@
...
+      {
+        "name": "increment score",
+        "type": "enum",
+        "enum": [
+          "INCR"
+        ],
+        "optional": true
+      },
+      {
+          "name": "payload",
```
Comment (oshadmi) [inline commands.json:371]: Same question on payload deprecation

File: commands.json [hunk 3]
```diff
@@ -30,5 +282,878 @@
...
+        "enum": [
+          "WITHSCORES"
+        ],
+        "optional": true
+      },
+      {
+        "name": "withpayloads",
+        "type": "enum",
+        "enum": [
+          "WITHPAYLOADS"
```
Comment (oshadmi) [inline commands.json:412]: Same question on payload deprecation

File: commands.json [hunk 4]
```diff
@@ -30,5 +282,878 @@
...
+      {
+        "name": "withsortkeys",
+        "type": "enum",
+        "optional": true,
+        "enum": [
+          "WITHSORTKEYS"
+        ]
+      },
+      {
+        "name": "withpatloads",
```
Comment (oshadmi) [inline commands.json:695]:
```suggestion
        "name": "withpayloads",
```

File: commands.json [hunk 5]
```diff
@@ -30,5 +282,878 @@
...
+        "type": "enum",
+        "optional": true,
+        "enum": [
+          "WITHPAYLOADS"
```
Comment (oshadmi) [inline commands.json:699]: Same question on payload deprecation

File: commands.json [hunk 6]
```diff
@@ -30,5 +282,878 @@
...
+        "enum": [
+          "WITHSCORES"
+        ]
+      },
+      {
+        "name": "withpatloads",
+        "type": "enum",
+        "optional": true,
+        "enum": [
+          "WITHPAYLOADS"
```
Comment (oshadmi) [inline commands.json:683]: Same question on payload deprecation

File: commands.json [hunk 7]
```diff
@@ -30,5 +282,878 @@
...
+        "multiple": true,
+        "block": [
+          {
+            "name": "numeric_field",
+            "type": "string",
+            "command":"FILTER"
+          },
+          {
+            "name": "min",
+            "type": "double"
```
Comment (oshadmi) [inline commands.json:715]: Need to also mention `-inf` (use a more complex enum?)

File: commands.json [hunk 8]
```diff
@@ -30,5 +282,878 @@
...
+          },
+          {
+            "name": "max",
+            "type": "double"
```
Comment (oshadmi) [inline commands.json:719]: Need to also mention `+inf` (use a more complex enum?)

File: commands.json [hunk 9]
```diff
@@ -30,5 +282,878 @@
...
+      {
+        "name": "summarize",
+        "type": "block",
+        "optional": true,
+        "block": [
+          {
+            "name": "summerize",
+            "type": "enum",
+            "enum": [
+              "SUMMERIZE"
```
Comment (oshadmi) [inline commands.json:815]:
```suggestion
              "SUMMARIZE"
```

File: commands.json [hunk 10]
```diff
@@ -30,5 +282,878 @@
...
+          }
+        ]
+      },
+      {
+        "name": "summarize",
+        "type": "block",
+        "optional": true,
+        "block": [
+          {
+            "name": "summerize",
```
Comment (oshadmi) [inline commands.json:812]:
```suggestion
            "name": "summarize",
```

File: commands.json [hunk 11]
```diff
@@ -30,5 +282,878 @@
...
+              },
+              {
+                "name": "field",
+                "type": "string",
+                "multiple": true
+              }
+            ]
+          },
+          {
+            "name": "tags",
```
Comment (oshadmi) [inline commands.json:885]: Shouldn't be an arg in the `fields` block?

File: commands.json [hunk 12]
```diff
@@ -30,5 +282,878 @@
...
+        ]
+      },
+      {
+        "name": "slop",
+        "type": "integer",
+        "optional": true,
+        "command":"SLOP"
+      },
+      {
+        "name": "tags",
```
Comment (oshadmi) [inline commands.json:911]: Should be in a block together with `SLOP`?

File: commands.json [hunk 13]
```diff
@@ -30,5 +282,878 @@
...
+        "optional": true,
+        "enum": [
+          "EXPLAINSCORE"
+        ]
+      },
+      {
+        "name": "payload",
+        "type": "string",
+        "optional": true,
+        "command":"PAYLOAD"
```
Comment (oshadmi) [inline commands.json:948]: Same question on payload deprecation

File: commands.json [hunk 14]
```diff
@@ -30,5 +282,878 @@
...
+      {
+        "name": "index",
+        "type": "key"
+      },
+      {
+        "name": "query",
+        "type": "string"
+      },
+      {
+          "name": "dist",
```
Comment (oshadmi) [inline commands.json:508]: Better use "distance" (in case it is visible to users in some tool tip, etc.)

File: commands.json [hunk 15]
```diff
@@ -30,5 +282,878 @@
...
+          {
+            "name": "limit",
+            "type": "enum",
+            "enum": [
+              "LIMIT"
+            ]
+          },
+          {
+            "name": ["offset", "num"],
+            "type": ["string", "string"]
```
Comment (oshadmi) [inline commands.json:984]: type of `num` is string or integer?

File: commands.json [hunk 16]
```diff
@@ -30,5 +282,878 @@
...
+      },
+      {
+        "name": "apply",
+        "type": "block",
+        "optional": true,
+        "multiple": true,
+        "block" : [
+          {
+            "name": "expression",
+            "type": "integer",
```
Comment (oshadmi) [inline commands.json:1121]:
```suggestion
            "type": "string",
```

File: commands.json [hunk 17]
```diff
@@ -30,5 +282,878 @@
...
+          {
+            "name": "limit",
+            "type": "enum",
+            "enum": [
+              "LIMIT"
+            ]
+          },
+          {
+            "name": ["offset", "num"],
+            "type": ["string", "string"]
```
Comment (oshadmi) [inline commands.json:1145]: type of `num` is `integer`?

**Review by oshadmi — COMMENTED** [review]
Some comments/questions

File: commands.json [hunk 18]
```diff
@@ -1,11 +1,13 @@
 {
   "FT.CREATE": {
     "summary": "Creates an index with the given spec",
-    "complexity": "O(1)",
+    "complexity": [
+      "O(K) at creation where K is the number of fields,",
+      "O(N) if scanning the keyspaceis triggered where N is the number of keys in the keyspace"],
```
Comment (oshadmi) [inline commands.json:6]: > O(N) if scanning the keyspaceis

Is it optional?

```suggestion
      "O(N) if scanning the keyspace is triggered, where N is the number of keys in the keyspace"],
```

File: commands.json [hunk 19]
```diff
@@ -562,27 +564,27 @@
...
       }
     ],
-    "since": "1.0.0",
+    "since": "1.4.0",
     "group": "search"
   },
   "FT.DICTDUMP": {
     "summary": "Dumps all terms in the given dictionary",
-    "complexity": "O(1)",
+    "complexity": "O(N)",
```
Comment (oshadmi) [inline commands.json:580]:
```suggestion
    "complexity": "O(N), where N is the size of the dictionary",
```

File: commands.json [hunk 20]
```diff
@@ -265,11 +267,11 @@
   },
   "FT.DROPINDEX": {
     "summary": "Deltes the index",
-    "complexity": "O(N)",
+    "complexity": "O(1) or O(N) if documents are removed",
```
Comment (oshadmi) [inline commands.json:270]:
```suggestion
    "complexity": "O(1) or O(N) if documents are deleted, where N is the number of keys in the keyspace",
```

File: commands.json [hunk 21]
```diff
@@ -6,21 +6,273 @@
...
+        "optional": true
+      },
+      {
+        "name": "score_attribute",
+        "type": "string",
+        "command":"SCORE_FIELD",
+        "optional": true
+      },
+      {
+        "name": "payload_attribute",
```
[the remark this answers is not in the document]
Comment (ashtul) [reply commands.json:62]: @oshadmi  How?

[the remark this answers is not in the document]
Comment (oshadmi) [reply commands.json:62]: Seems that currently only possible in text, as in [here](https://github.com/redis/redis-doc/blob/8678b903ca5e2d4f8afe1478326ddd89e2c35eed/commands.json#L3966)

File: commands.json [hunk 22]
```diff
@@ -6,21 +6,273 @@
...
+            ],
+            "GEO": [
+              {
+                "name": "type",
+                "type": "string"
+                
+              }
+            ],
+            "enum": [
+              "TEXT", "TAG", "NUMERIC", "GEO"
```
[the remark this answers is not in the document]
Comment (ashtul) [reply commands.json:173]: After a discussion, we opted for enum for field type w/o sub flags

[the remark this answers is not in the document]
Comment (ashtul) [reply commands.json:173]: The current syntax does not support complex enums

File: commands.json (hunk 12 above)
Comment (oshadmi) [reply commands.json:917]: OK, should not be in another block.

File: commands.json [hunk 23]
```diff
@@ -30,5 +288,878 @@
...
+      },
+      {
+        "name": "load",
+        "type": "block",
+        "optional": true,
+        "block": [
+          {
+            "name": "count",
+            "type": "string",
+            "command":"LOAD"
```
Comment (oshadmi) [inline commands.json:1027]: What about `LOAD *`? Do we want to add this option? (either `LOAD *` or `LOAD count field [field...]`)
Could be useful here, although currently they wouldn't be mutual exclusive.

**Review by oshadmi — COMMENTED** [review]
Few more comments

File: commands.json (hunk 8 above)
Comment (ashtul) [reply commands.json:725]: Yes.
Due to the current limitations, lets keep it as is.

File: commands.json (hunk 23 above)
Comment (ashtul) [reply commands.json:1027]: Let's wait with it to the new syntax with `oneof` option.

**Review by oshadmi — APPROVED** [review]
Thanks 👍🏼

**Pull request merged**